src/components/datagrid/utils/focus.ts

```diff
@@ -67,10 +67,21 @@ export const useFocus = ({
     EuiDataGridFocusedCell | undefined
   >(undefined);
 
-  const setFocusedCell = useCallback((focusedCell: EuiDataGridFocusedCell) => {
-    _setFocusedCell(focusedCell);
-    setIsFocusedCellInView(true); // scrolling.ts ensures focused cells are fully in view
-  }, []);
+  const setFocusedCell = useCallback(
+    (nextFocusedCell: EuiDataGridFocusedCell) => {
+      if (
+        focusedCell &&
+        nextFocusedCell[0] === focusedCell[0] &&
+        nextFocusedCell[1] === focusedCell[1]
+      ) {
+        return;
+      }
```

Comment:
Small syntax nit - let's use optional chaining here instead of the first `focusedCell &&` - it's a little easier to read just IMO. It would also be nice to add a comment here explaining the early return

```suggestion
      // If the x/y coordinates remained the same, don't update
      if (
        nextFocusedCell[0] === focusedCell?.[0] &&
        nextFocusedCell[1] === focusedCell?.[1]
      ) {
        return;
      }
```